x/mongo/driver/compression.go

```diff
@@ -41,6 +40,50 @@ func getZstdEncoder(l zstd.EncoderLevel) (*zstd.Encoder, error) {
 	return encoder, nil
 }
 
+var zlibEncoders = &sync.Map{}
```

Comment:
IIUC the pointer to this map is unecessary.
```suggestion
var zlibEncoders = sync.Map
```